frontend/src/container/Sidebar.jsx

```diff
@@ -79,18 +79,27 @@ export class Sidebar extends React.Component {
         <WorkspaceListItem
           activeWorkspaceId={state.activeWorkspaceId}
           allowedAppList={space.sidebarEntryList}
+          hasChildren={space.children.length > 0}
+          hideChildren={state[`hideChildrenOf${space.id}`]}
           label={space.label}
           level={spaceLevel}
           onClickAllContent={this.handleClickAllContent}
           userRoleIdInWorkspace={findUserRoleIdInWorkspace(props.user.userId, space.memberList, ROLE_LIST)}
           workspaceId={space.id}
           id={this.spaceItemId(space.id)}
+          onToggleHideChildren={() => this.handleToggleHideChildren(space.id)}
         />
-        {space.children.length !== 0 && this.displaySpace(spaceLevel + 1, space.children)}
+        {!state[`hideChildrenOf${space.id}`] &&
+          space.children.length !== 0 &&
+          this.displaySpace(spaceLevel + 1, space.children)}
       </React.Fragment>
     )
   }
 
+  handleToggleHideChildren = (id) => {
+    this.setState(prev => ({ [`hideChildrenOf${id}`]: !prev[`hideChildrenOf${id}`] }))
```

Comment:
I'm not convinced of using the `state` object as a dictionary: objects should have a fixed list of properties.
Wouldn't an array of folded/hidden space ids + `this.state.foldedSpaces.includes(space.id)` in the `render()` method work?

Reply:
An object in the state would work too (and be probably a bit easier to use than an array.

Reply:
The array I can see how it could work, the object I'm not sure. Can you explain a little bit more your idea?